web/src/people.js

```diff
@@ -1282,6 +1282,30 @@ export function report_late_add(user_id, email) {
     }
 }
 
+function make_user(partial_user) {
+    return {
+        user_id: partial_user.user_id ?? -1,
+        email: partial_user.email ?? "",
+        full_name: partial_user.full_name ?? "",
+        is_active: partial_user.is_active ?? false,
+        is_admin: partial_user.is_admin ?? false,
+        is_guest: partial_user.is_guest ?? false,
+        timezone: partial_user.timezone ?? "",
+        is_bot: partial_user.is_bot ?? false,
+        avatar_version: partial_user.avatar_version ?? 0,
+        date_joined: partial_user.date_joined ?? "",
+        delivery_email: partial_user.delivery_email ?? "",
+        profile_data: partial_user.profile_data ?? {},
+        is_billing_admin: partial_user.is_billing_admin ?? false,
+        role: partial_user.role ?? 0,
+        is_owner: partial_user.is_owner ?? false,
+        avatar_url: partial_user.avatar_url ?? "",
+        bot_owner_id: partial_user.bot_owner_id ?? undefined,
+        bot_type: partial_user.bot_type ?? 0,
+        is_moderator: partial_user.is_moderator ?? false,
+    };
```

Comment:
I think a number of these defaults are dangerous -- for example, `0` is not a valid possible value for `role`. My current thinking is we should rewrite this function to accept just the 3 parameters that we call it with, and pick a consistent set of values that are the defaults for a newly created user, and perhaps an extra new field `is_missing_server_data` that we can use to keep track of the fact that this is a fake user object.